llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -202,7 +202,9 @@ std::string manglePrimitiveType(const Type *T) {
   if (T->isStructTy())
     return T->getStructName().str();
   if (T->isArrayTy())
-    return "A" + manglePrimitiveType(T->getArrayElementType());
+    return "A" + std::to_string(T->getArrayNumElements()) +
+           "_" + manglePrimitiveType(T->getArrayElementType());
```

Comment:
BTW, "primitive" in "manglePrimitiveType" is no longer valid, as we are mangling quite non-primitive ones. Would be good to change.

Reply:
Renamed to `mangleType` in c88f6ffc433b324c3e3359d4887cc1c027ff2c64